### DISCUSSION TITLE
How to initialize database dependency for unit tests

### DISCUSSION THREAD

**Discussion by mplorentz** [pointfreeco/swift-dependencies#134] [Q&A] [answered]
I'm having trouble with what I think should be a simple task for a dependency injection library. I feel like I am missing something.

My app uses Core Data and I have created a `PersistenceController` class that sets up the Core Data stack and manages various `NSManagedObjectContext`s. This persistence controller is managed by the Dependencies library and I have configured a `testValue` that uses an in-memory persistent store.

**How can I create a fresh `PersistenceController` dependency for each test in an XCTestCase subclass?** 

I know I could wrap every test in a `withDependencies(:operation:)` block, but that's a lot of duplicated code, so I'm looking for a cleaner way. I tried abusing `withEscapedDependencies()` to set up a new `PersistenceController` in `setUp()` and not yield the continuation until `tearDown()` but that didn't seem to work either.

**Comment by mbrandonw** [comment, marked as answer]
Hi @mplorentz, I'm not super familiar with the specifics of CoreData anymore, but there's no need to use `withDependencies` in each individual test method. You can do it once for all tests in a class by overriding `invokeTest`. We have some docs here:

https://github.com/pointfreeco/swift-dependencies/blob/b323c53817e4936469309df8dbb7bf01a385f220/Sources/Dependencies/Documentation.docc/Articles/OverridingDependencies.md?plain=1#L127-L134

**Reply by mplorentz** [reply]
Thank you so much! This is exactly what I needed. I skimmed that docs page several times and somehow didn't see that part 😬.

**Reply by acosmicflamingo** [reply]
Geez...how did I miss that? 🤩
